lib/crowbar/client/command/proposal/edit.rb

```diff
@@ -140,6 +150,29 @@ def payload_content
               from_editor
             end
           end
+
+          def check_nodes(configuration)
+            avaliable_nodes = JSON.parse(Request::Node::List.new({}).process)
```

Comment:
This part (including following lines) can be simplified using `valid_elements` from https://github.com/crowbar/crowbar-client/blob/master/lib/crowbar/client/mixin/proposal.rb#L31
It returns list of valid names for "elements" section. It also supports clusters (i.e. `cluster:<name>`). Clusters also make the name `available_nodes` not fully correct.